src/core/optimizer_stats.jl

```diff
@@ -67,42 +63,36 @@ end
 """
 Convert OptimizerStats to an array of floats that can be serialized to HDF5.
 """
-function to_array(stats::OptimizerStats)
-    return [Float64(getfield(stats, x)) for x in fieldnames(OptimizerStats)]
+function to_array(stats::T) where {T <: OptimizerStats}
+    field_values = Vector{Float64}(undef, fieldcount(T))
+    for (ix, field) in enumerate(fieldnames(T))
+        value = ismissing(getfield(stats, field))
+        field_values[ix] = ismissing(value) ? NaN : value
+    end
+    return field_values
 end
 
 function to_dataframe(stats::OptimizerStats)
     df = DataFrames.DataFrame([to_namedtuple(stats)])
-    if !_part_of_simulation(stats)
-        DataFrames.select!(df, _BASE_FIELDS)
-    end
+    #if !_part_of_simulation(stats)
```

Comment:
I commented this out because I wasn't sure if it will be needed anymore. There are several parts of optimizer_stats that we should delete if we confirm aren't needed.